### PR TITLE
Fixing defects on tickets 6233, 6342

### LINKED ISSUE #6233: Need to display the unit (load based indicator, CO, CCO, etc.), unit program and reporting frequency data in the UI.
ECMPS 1.0 has the following read-only fields in the Unit Characteristics section that need to be added to 2.0. A new Unit Program and Reporting Frequency grid needs to be added to the "Unit Information" section of the "Monitoring Plans" screen. Adding the editable Non Load Based Indicator field will be handled in #6342:

* Unit Attributes (Data is from the CAMD.UNIT table):
     * Non Load Based Indicator
     * Source Category Code
     * Commence Operation Date (CO)
     * Commercial Operation Date (CCO)
     * Unit Type Code
     * Operation Status Code
     * Operation Status Begin Date
   
image

* Unit Program (Data is from the CAMD.UNIT_PROGRAM table):
     * Program Code
     * Unit Classification
     * Unit Monitor Certification Begin Date
     * Unit Monitor Certification Deadline
     * Emissions Recording Begin Date
     * End Date

image

* Reporting Frequency (Data is from the CAMDECMPS.MONITOR_PLAN_REPORTING_FREQ table):
     * Reporting Frequency
     * Monitoring Plan Locations
     * Begin Quarter
     * End Quarter

image

## Acceptance Tests

- A Unit Attributes grid is added to the Unit Information section of the Monitoring Plans screen with the Read-Only fields listed above
- A Unit Program grid is added to the Unit Information section of the Monitoring Plans screen with the Read-Only fields listed above
- A Reporting Frequency grid is added to the Unit Information section of the Monitoring Plans screen with the Read-Only fields listed above

### LINKED ISSUE #6342: Add Non Load Based Indicator field to UI
## Overview

For ECMPS 1.0, the Non-Load-Based Indicator (NON_LOAD_BASED_IND) is stored in the CAMD.UNIT table on the host and the ECMPS.UNIT table on the client.  Non-Load-Based Indicator is editable in ECMPS 1.0 and should be editable in ECMPS 2.0.

Currently ECMPS 2.0 stores the Non-Load-Based Indicator in the CAMD.UNIT table.  However, editable (submittable) data must exist in the CAMDECMPSWKS schema when it is edited and before it is submitted.  To enable the editing and submitting of the Non-Load-Based Indicator,  a new UNIT table will be created in the CAMDECMPSWKS schema that will contain the columns listed below.  The Non-Load-Based Indicator in that table will be submitted to the CAMD.UNIT table when an MP is submitted, and will be updated from the CAMD.UNIT table when an MP is reverted (or otherwise copied) from official.

## Acceptance Tests

- A Non Load Based Indicator field is present in the Unit Attributes grid and can be edited in the workspace.
     - The Non-Load-Based Indicator field is a dropdown field that can only be either 0 or 1.
     - Reverting the Monitoring Plan to Official Record removes the edits made to the Non Load Based Indicator field
     - Submitting a monitoring plan with edits to the Non Load Based Indicator updates the monitoring plan in Official view

## Change List

- [ ] Add a UNIT table to the CAMDECMPSWKS schema using the structure below.
- [ ] Add a "Unit Attributes" grid to the "Unit Information" section of the "Monitoring Plans" screen.
      1. The grid should contain both an Actions and a Non Load Based Indicator column.
      2. The Action column should contain a link to open a Unit Attribute screen in either view or view/edit mode.
      3. The Unit Attribute screen should show and in view/edit mode allow the editing of the Non Load Based Indicator.
- [ ] Include the update of the NON_LOAD_BASED_IND in the CAMDECMPSWKS.UNIT table when a monitoring plan is reverted or loaded into the workspace.
- [ ] Include the update of the NON_LOAD_BASED_IND in the CAMDECMPS.UNIT table when a monitoring plan is submitted.
- [ ] Ensure that the CAMDECMPS.UNIT.NON_LOAD_BASED_IND defaults to 0 (zero) when created.

## Additional Considerations

#6233 was created to display Unit Attributes including Non Load Based Indicator, CO, CCO, etc, as well as Unit Program and Reporting Frequency data.  This matches the ECMPS 1.0 Unit Characteristics Tab's sub tabs:

- **Unit Attributes**:
  - Non Load Based Indicator
  - Source Category Code
  - Commence Operation Date (CO)
  - Commercial Operation Date (CCO)
  - Unit Type Code
  - Operation Status Code
  - Operation Status Begin Date
- **Unit Capacity**: Already implemented.
- **Unit Control**: Already Implemented
- **Unit Fuel**: Already Implemented
- **Unit Program**
- **Reporting Frequency**

Based on this ticket, #6233 should include the items under bullet **Unit Attributes** above in the grid and pop-up for this ticket as read-only items.  **Pull the additional (non-editable) values from the CAMD.UNIT table.  Do not add them to the CAMDECMPSAUX table.**

## CAMDECMPSWKS.UNIT Table

| Column Name | Data Type | Required | Constraining Values |
| :--- | :--: | :--: | :--: |
| UNIT_ID | numeric(38) | Yes |  |
| NON_LOAD_BASED_IND | numeric(1) | Yes | 0 or 1 |
| USER_ID | varchar(160) | Yes |  |
| ADD_DATE | timestamp | Yes |  |
| UPDATE_DATE | timestamp | No |  |

## Original Description

**_See also #6233_**

This is a placeholder for the needed changes to add the ability for users to update the Non Load Based Indicator. This field does not currently exist and can only be seen in a MP Printout Report.

The database column (NON_LOAD_BASED_IND) is currently in CAMD.UNIT and needs to be moved to CAMDECMPS... (separate ticket(s) for db changes?).

Field in ECMPS 1.0:

image

### PR DESCRIPTION
**Pull request by yonatan-dp**
Fixing defects on tickets 6233, 6342

### CODE REVIEW

**Review by djw4erg — CHANGES_REQUESTED** [review]
An insert from the **CAMDECMPSWKS** **UNIT** table into the **CAMD** **UNIT** table should never occur.  Each of these scripts for the copy_monitor_plan_from_workspace_to_global should **only perform updates** of the following:

- NON_LOAD_BASED_IND
- USERID
- UPDATE_DATE

Additionally, the changes should be compiled and tested in the Dev database before a merge can occur, and before the PR if possible.

**Comment by yonatan-dp** [comment]
> An insert from the **CAMDECMPSWKS** **UNIT** table into the **CAMD** **UNIT** table should never occur. Each of these scripts for the copu_monitor_plan_from_workspace_to_global should **only perform updates** of the following:
> 
> * NON_LOAD_BASED_IND
> * USERID
> * UPDATE_DATE
> 
> Additionally, the changes should be compiled and tested in the Dev database before a merge can concern, and before PR if possible.

Updated per comment.

File: camdecmps/procedures/copy_monitor_plan_from_workspace_to_global.sql [hunk 1]
```diff
@@ -198,30 +198,16 @@ BEGIN
...
-        source_category_cd = EXCLUDED.source_category_cd,
-        naics_cd = EXCLUDED.naics_cd,
-        no_active_gen_ind = EXCLUDED.no_active_gen_ind,
-        non_load_based_ind = EXCLUDED.non_load_based_ind,
-        actual_90th_op_date = EXCLUDED.actual_90th_op_date,
-        moved_ind = EXCLUDED.moved_ind,
-        userid = EXCLUDED.userid,
-        add_date = EXCLUDED.add_date,
-        update_date = EXCLUDED.update_date;
+    --We cannot insert a copy of the camdecmpswks.unit data because the two tables have different columns.
```
Comment (djw4erg) [inline camdecmps/procedures/copy_monitor_plan_from_workspace_to_global.sql]: Sorry about this, but this comment is not correct.  It should be something like:

> The data columns other than NON_LOAD_BASED_IND are managed by the CBS application in the CAMD schema.
> Only the NON_LOAD_BASED_IND column and USERID and UPDATE_DATE should be updated when copying from CAMDECMPSWKS to CAMD, and only when a row already exists in the CAMD.UNIT table.

File: deployments/ecmps/release-v2.0-fix/Sprint10/5995/camdecmps/copy_monitor_plan_from_workspace_to_global.sql [hunk 2]
```diff
@@ -198,30 +198,16 @@ BEGIN
...
-        source_category_cd = EXCLUDED.source_category_cd,
-        naics_cd = EXCLUDED.naics_cd,
-        no_active_gen_ind = EXCLUDED.no_active_gen_ind,
-        non_load_based_ind = EXCLUDED.non_load_based_ind,
-        actual_90th_op_date = EXCLUDED.actual_90th_op_date,
-        moved_ind = EXCLUDED.moved_ind,
-        userid = EXCLUDED.userid,
-        add_date = EXCLUDED.add_date,
-        update_date = EXCLUDED.update_date;
+    --We cannot insert a copy of the camdecmpswks.unit data because the two tables have different columns.
```
Comment (djw4erg) [inline deployments/ecmps/release-v2.0-fix/Sprint10/5995/camdecmps/copy_monitor_plan_from_workspace_to_global.sql]: Sorry about this, but this comment is not correct.  It should be something like:

> The data columns other than NON_LOAD_BASED_IND are managed by the CBS application in the CAMD schema.
> Only the NON_LOAD_BASED_IND column and USERID and UPDATE_DATE should be updated when copying from CAMDECMPSWKS to CAMD, and only when a row already exists in the CAMD.UNIT table.

File: deployments/ecmps/release-v2.0-fix/Sprint10/6233/camdecmps/copy_monitor_plan_from_workspace_to_global.sql [hunk 3]
```diff
@@ -197,31 +197,17 @@ BEGIN
...
-         naics_cd = EXCLUDED.naics_cd,
-         no_active_gen_ind = EXCLUDED.no_active_gen_ind,
-         non_load_based_ind = EXCLUDED.non_load_based_ind,
-         actual_90th_op_date = EXCLUDED.actual_90th_op_date,
-         moved_ind = EXCLUDED.moved_ind,
-         userid = EXCLUDED.userid,
-         add_date = EXCLUDED.add_date,
-         update_date = EXCLUDED.update_date;
+    -- UNIT --
+    --We cannot insert a copy of the camdecmpswks.unit data because the two tables have different columns.
```
Comment (djw4erg) [inline deployments/ecmps/release-v2.0-fix/Sprint10/6233/camdecmps/copy_monitor_plan_from_workspace_to_global.sql]: Sorry about this, but this comment is not correct.  It should be something like:

> The data columns other than NON_LOAD_BASED_IND are managed by the CBS application in the CAMD schema.
> Only the NON_LOAD_BASED_IND column and USERID and UPDATE_DATE should be updated when copying from CAMDECMPSWKS to CAMD, and only when a row already exists in the CAMD.UNIT table.

**Review by djw4erg — CHANGES_REQUESTED** [review]

File: camdecmps/procedures/copy_monitor_plan_from_workspace_to_global.sql (hunk 1 above)
Comment (yonatan-dp) [reply camdecmps/procedures/copy_monitor_plan_from_workspace_to_global.sql]: Updated comment (comment was from the original ticket where we only had selected columns in the camdecmpswks.unit table)

File: deployments/ecmps/release-v2.0-fix/Sprint10/5995/camdecmps/copy_monitor_plan_from_workspace_to_global.sql (hunk 2 above)
Comment (yonatan-dp) [reply deployments/ecmps/release-v2.0-fix/Sprint10/5995/camdecmps/copy_monitor_plan_from_workspace_to_global.sql]: Updated comment (comment was from the original ticket where we only had selected columns in the camdecmpswks.unit table)

File: deployments/ecmps/release-v2.0-fix/Sprint10/6233/camdecmps/copy_monitor_plan_from_workspace_to_global.sql (hunk 3 above)
Comment (yonatan-dp) [reply deployments/ecmps/release-v2.0-fix/Sprint10/6233/camdecmps/copy_monitor_plan_from_workspace_to_global.sql]: Updated comment (comment was from the original ticket where we only had selected columns in the camdecmpswks.unit table)

**Review by djw4erg — APPROVED** [review]

**Pull request merged**